src/numerics/petsc_matrix.C

```diff
@@ -1256,7 +1256,45 @@ T PetscMatrix<T>::operator () (const numeric_index_type i_in,
   return value;
 }
 
+template <typename T>
+void PetscMatrix<T>::get_row (numeric_index_type i_in,
+                              std::vector<numeric_index_type> & indices,
+                              std::vector<T> & values) const
+{
+  libmesh_assert (this->initialized());
+
+  // PETSc 2.2.1 & newer
```

Comment:
I'm guessing this comment is copy-pasted and you didn't really test with PETSc 2.2.1 ;-)